hbase-server/src/main/java/org/apache/hadoop/hbase/wal/BoundedRecoveredHFilesOutputSink.java

```diff
@@ -99,12 +102,13 @@ public void append(RegionEntryBuffer buffer) throws IOException {
       }
     }
 
-    // The key point is create a new writer for each column family, write edits then close writer.
+    // Create a new hfile writer for each column family, write edits then close writer.
     String regionName = Bytes.toString(buffer.encodedRegionName);
     for (Map.Entry<String, CellSet> cellsEntry : familyCells.entrySet()) {
       String familyName = cellsEntry.getKey();
       StoreFileWriter writer = createRecoveredHFileWriter(buffer.tableName, regionName,
         familySeqIds.get(familyName), familyName, isMetaTable);
+      LOG.trace("Created {}", writer.getPath());
```

Comment:
Good to guard with `LOG.isTraceEnabled()`?

Reply:
Why you say? The slf4j doesn't do any work creating log string if we are not at log TRACE level.

Reply:
This is to avoid the cost of parameter construction: http://logging.apache.org/log4j/1.2/manual.html#performance

Reply:
My bad, this is old doc for apache log4j, not slf4j

Reply:
You are right, I just looked into one of the implementors of slf4j (Log4jLoggerAdapter) and it already covers `isTraceEnabled` check:

```
    public void trace(String format, Object arg) {
        if (isTraceEnabled()) {
            FormattingTuple ft = MessageFormatter.format(format, arg);
            logger.log(FQCN, traceCapable ? Level.TRACE : Level.DEBUG, ft.getMessage(), ft.getThrowable());
        }
    }
```

We are following explicit guarding for the cases of Trace and Debug level log at multiple places, all are redundant checks since library already does it for us.

Reply:
Thanks @virajjasani

Reply:
IIRC, this extra check could be used only if the parameter self will cause a heavy call.